File: Python/Merge Two Sorted Lists.py
# Merge two sorted linked lists and return it as a new list. The new list should be made by splicing together the nodes of the first two lists.

# Definition for singly-linked list.
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution:
    # @param two ListNodes
    # @return a ListNode
    def mergeTwoLists(self, l1, l2):
		ls1 = l1
		ls2 = l2
		head = ListNode(-1)
		p =head
		if l1 == None:
			return l2
		if l2 == None:
			return l1
		while ls1 != None and ls2 != None:
			if ls1.val < ls2.val :
				p.next = ls1
				ls1 = ls1.next
			else:
				p.next = ls2
				ls2 = ls2.next
			p = p.next
		if ls1 != None:
			p.next = ls1
		if ls2 != None:
			p.next = ls2
		p = head.next
		return p

File: Java/Merge Two Sorted Lists.java
class Merge Two Sorted Lists{
	public ListNode mergeTwoLists(ListNode l1, ListNode l2) {
        if(l1 == null)
        	return l2;
        if(l2 == null)
        	return l1;
        ListNode dummy = new ListNode(-1);
        dummy.next = l1;
        ListNode h1 = l1;
        ListNode h2 = l2;
        ListNode pre = dummy;
        while(h1 != null && h2 != null){
        	if(h1.val <= h2.val){
        		h1 = h1.next;
        	}
        	else{
        		ListNode temp = h2.next;
        		pre.next = h2;
        		h2.next = h1;
        		h2 = temp;
        		pre = pre.next;
        	}
        }
        if(h2 != null)
        	pre.next = h2;
        return dummy.next;
    }
}